tests/aws/services/cloudformation/resources/test_sns.py

```diff
@@ -191,3 +191,166 @@ def test_sns_subscription_region(
     sub_arn = stack.outputs["SubscriptionArn"]
     subscription = sns_client.get_subscription_attributes(SubscriptionArn=sub_arn)
     snapshot.match("subscription-1", subscription)
+
+
+@markers.aws.validated
+@markers.snapshot.skip_snapshot_verify(
+    paths=[
+        "$..Attributes.DeliveryPolicy",
+        "$..Attributes.EffectiveDeliveryPolicy",
+        "$..Attributes.Policy.Statement..Action",  # SNS:Receive is added by moto but not returned in AWS
+    ]
+)
+def test_sns_topic_update_attributes(deploy_cfn_template, aws_client, snapshot):
+    """Test updating SNS Topic DisplayName and Tags."""
+    snapshot.add_transformer(snapshot.transform.key_value("TopicArn"))
+    snapshot.add_transformer(
+        snapshot.transform.key_value(
+            "SubscriptionArn", "PendingConfirmation", reference_replacement=False
+        ),
+    )
+
+    topic_name = f"test-topic-{short_uid()}"
+
+    stack = deploy_cfn_template(
+        parameters={
+            "TopicName": topic_name,
+            "DisplayName": "Initial Display Name",
+            "Environment": "test",  # tag
+            "Project": "localstack",  # tag
+        },
+        template_path=os.path.join(
+            os.path.dirname(__file__), "../../../templates/sns_topic_update.yaml"
+        ),
+    )
+
+    topic_arn = stack.outputs["TopicArn"]
+
+    initial_attrs = aws_client.sns.get_topic_attributes(TopicArn=topic_arn)
+    snapshot.match("initial-topic-attributes", initial_attrs)
+
+    initial_tags = aws_client.sns.list_tags_for_resource(ResourceArn=topic_arn)
+    tag_dict = {tag["Key"]: tag["Value"] for tag in initial_tags["Tags"]}
+    assert tag_dict["Environment"] == "test"
+    assert tag_dict["Project"] == "localstack"
```

Comment:
```suggestion
    snapshot.match("initial-tags", initial_tags)
```

We may need to add specific non-supported tags to the `skip_snapshot_verify` decorator at the top of the test function, as CFn adds its own tags to resources and we don't support that yet.